ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessPackageUpload.scala

```diff
@@ -133,3 +138,7 @@ private[kvutils] case class ProcessPackageUpload(
   }
 
 }
+
+object ProcessPackageUpload{
+  val serialContext = ExecutionContext.fromExecutor(Executors.newSingleThreadExecutor())
```

Comment:
This executor and its thread will keep on living until the JVM shuts down. This is probably not a problem, but if you only need to run one task on a separate thread you may want to consider either shutting down the executor once its done or using a simple `Thread` and create a `Future` using a `Promise` based on that thread.

Reply:
What I need is all pre-loads to happen on the same single thread. Pre-loads can happen any time and they come in clusters. Doing them serially conserves resources and allows proper recognition of duplicates.